server/src/main/java/org/opensearch/common/settings/AbstractScopedSettings.java

```diff
@@ -121,8 +122,13 @@ protected AbstractScopedSettings(
                 keySettings.putIfAbsent(setting.getKey(), setting);
             }
         }
-        this.complexMatchers = Collections.unmodifiableMap(complexMatchers);
-        this.keySettings = Collections.unmodifiableMap(keySettings);
+        if (FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS)) {
```

Comment:
This is just adding support for dynamically registering settings during runtime. 
It has value for users with/without extensions, WDYT @reta @dblock ?

Reply:
@saratvemulapalli good question, I would say if we could split this pull request in two, dynamically registering settings during runtime + extenstion, than the feature flag could be gone for this part.

Reply:
Thanks @reta I can take care of that. Here is the original PR: https://github.com/opensearch-project/OpenSearch/pull/3753